### PR TITLE
Google Pay docs

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Add documentation on Google Pay + a migration guide for the new implementation

## Checklist 
- [x] PR is labelled 

COAND-941

### CODE REVIEW

File: docs/payment-methods/GOOGLE_PAY.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        setShippingAddressParameters(…)
+        setBillingAddressRequired(true)
+        setBillingAddressParameters(…)
+        setTotalPriceStatus("FINAL")
+        setGooglePayButtonStyling(…)
+    }
+}
+```
+
+`setSubmitButtonVisible` set to `true` to display the Google Pay button in the component. The default value is `false`. \
```
Comment (jreij) [inline docs/payment-methods/GOOGLE_PAY.md:68]: WDYT about making this a table?

File: docs/payment-methods/GOOGLE_PAY.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+`setMerchantAccount` sets the merchant account to be put in the payment token from Google to Adyen.\
+`setGooglePayEnvironment` sets the environment to be used by Google Pay.\
+`setMerchantInfo` sets the information about the merchant requesting the payment.\
+`setCountryCode` sets the ISO 3166-1 alpha-2 country code where the transaction is processed.\
+`setAllowedAuthMethods` sets the supported authentication methods.\
+`setAllowedCardNetworks` sets the allowed card networks. The allowed networks are automatically configured based on your account settings, but you can override them here.\
+`setAllowPrepaidCards` sets if you support prepaid cards.\
```
Comment (jreij) [inline docs/payment-methods/GOOGLE_PAY.md:75]: This should be `set to true if` or `set if`, not `sets if`

File: docs/payment-methods/GOOGLE_PAY.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+| `setBillingAddressRequired`        | Set to `true` if a billing address is required.                                                                                                   |
+| `setBillingAddressParameters`      | Allows to configure the billing address parameters.                                                                                               |
+| `setTotalPriceStatus`              | Sets the status of the total price used.                                                                                                          |
+| `setGooglePayButtonStyling`        | Allows to configure the styling of the Google Pay button.                                                                                         |
+
+## Migrating to 5.8.0+
+It is not necessary to migrate, but 5.8.0 introduced a simplified integration for Google Pay. This new integration among others gets rid of the deprecated `onActivityResult` and includes the Google Pay button. Follow the steps below to migrate from previous 5.x.x versions to 5.8.0:
+
+1. Remove deprecated Activity Result code
```
Comment (ozgur00) [inline docs/payment-methods/GOOGLE_PAY.md:92]: This line could be bold or somehow marked up to make it easier to read. Applies to line 118 and 138 as well.

**Review by araratthehero — APPROVED** [review]

File: example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/GooglePayViewModel.kt [hunk 4]
```diff
@@ -67,38 +67,40 @@ internal class GooglePayViewModel @Inject constructor(
...
-                merchantAccount = keyValueStorage.getMerchantAccount(),
-                shopperReference = keyValueStorage.getShopperReference(),
-                amount = keyValueStorage.getAmount(),
-                countryCode = keyValueStorage.getCountry(),
-                shopperLocale = keyValueStorage.getShopperLocale(),
-                splitCardFundingSources = keyValueStorage.isSplitCardFundingSources(),
-            ),
-        )
+    private suspend fun fetchPaymentMethods() =
+        @Suppress("RestrictedApi")
```
Comment (jreij) [inline example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/GooglePayViewModel.kt:71]: Can you add a todo to fix this later after we fix the dispatchers on develop and rebase?

**Review by ozgur00 — APPROVED** [review]

**Pull request merged**